packages/core/admin/admin/src/content-manager/components/BlocksInput/EditorLayout.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+
+import { Box, Flex, FocusTrap, Portal, IconButton, InputWrapper } from '@strapi/design-system';
+import { useLockScroll, pxToRem } from '@strapi/helper-plugin';
+import { Collapse } from '@strapi/icons';
+import { useIntl } from 'react-intl';
+import styled from 'styled-components';
+
+import { getTranslation } from '../../utils/translations';
+
+const CollapseIconButton = styled(IconButton)`
+  position: absolute;
+  bottom: ${pxToRem(12)};
+  right: ${pxToRem(12)};
+`;
+
+interface EditorLayoutProps {
+  children: React.ReactNode;
+  isExpandMode: boolean;
+  error?: string;
+  onCollapse: () => void;
+  disabled: boolean;
+  ariaDescriptionId: string;
+}
+
+const EditorLayout = ({
+  children,
+  isExpandMode,
+  error,
+  disabled,
+  onCollapse,
+  ariaDescriptionId,
+}: EditorLayoutProps) => {
+  const { formatMessage } = useIntl();
+
+  useLockScroll({ lockScroll: isExpandMode });
+
+  if (isExpandMode) {
+    return (
+      <Portal role="dialog" aria-modal={false}>
+        <FocusTrap onEscape={onCollapse} skipAutoFocus>
+          <ExpandWrapper
+            position="fixed"
+            top={0}
+            left={0}
+            right={0}
+            bottom={0}
+            zIndex={4}
+            justifyContent="center"
+            onClick={onCollapse}
+            data-testid="blocks-expand"
+          >
+            <Box
+              background="neutral0"
+              hasRadius
+              shadow="popupShadow"
+              overflow="hidden"
+              width="90%"
+              height="90%"
+              onClick={(e) => e.stopPropagation()}
+              aria-describedby={ariaDescriptionId}
+              position="relative"
+            >
+              <Flex height="100%" alignItems="flex-start" direction="column">
+                {children}
+                <CollapseIconButton
+                  aria-label={formatMessage({
+                    id: getTranslation('components.Blocks.collapse'),
+                    defaultMessage: 'Collapse',
+                  })}
+                  onClick={onCollapse}
+                >
+                  <Collapse />
+                </CollapseIconButton>
+              </Flex>
+            </Box>
+          </ExpandWrapper>
+        </FocusTrap>
+      </Portal>
+    );
+  }
+
+  return (
+    <InputWrapper
+      direction="column"
+      alignItems="flex-start"
+      height="512px"
+      disabled={disabled}
+      hasError={Boolean(error)}
+      style={{ overflow: 'hidden' }}
+      aria-describedby={ariaDescriptionId}
+      position="relative"
+    >
+      {children}
+    </InputWrapper>
+  );
+};
+
+const ExpandWrapper = styled(Flex)`
+  // Background with 10% opacity
+  background: ${({ theme }) =>
+    `${theme.colors.neutral900}${Math.floor(0.1 * 255)
+      .toString(16)
+      .padStart(2, '0')}`};
+`;
```

Comment:
- could you move this above the main component? We usually put styled components near the top
- you can use #rrggbbaa [like in the design system](https://github.com/strapi/design-system/blob/main/packages/strapi-design-system/src/ModalLayout/ModalLayout.tsx#L17):

```
background: ${({ theme }) => `${theme.colors.neutral800}1F`};
```

it also matches the design: neutral800 with 20% opacity

Reply:
thanks